### PR TITLE
[IMP]HR_HOLIDAY:In employee form view in rename Approved Leaves oe_stat ...

### PR DESCRIPTION
**Pull request by ara-odoo**
...button into Leaves to process also add a TO Do filter and rename Number Of Leaves into Remaining Leaves oe_stat button and all group visible by employee ,add a filter type and year. also change a oe_stat button on number of leaves.
https://www.odoo.com/web?#id=10204&view_type=form&model=project.task&action=327
pad: https://pad.odoo.com/p/r.2FBjWajDmTJjrTiC

### CODE REVIEW

File: addons/hr_holidays/hr_holidays_view.xml [hunk 1]
```diff
@@ -587,19 +587,22 @@
...
+                    <field name="show_leaves" invisible="1"/>
                     <button name="%(act_hr_employee_holiday_request_approved)d"
                             icon="fa-calendar"
                             class="oe_stat_button"
                             type="action"
-                            groups="base.group_hr_user">
-                        <field name="approved_leaves_count" widget="statinfo"/>
+                            attrs="{'invisible': [('show_leaves','=', False)]}"
+                            groups="base.group_user" help="Leaves to Process">
+                        <field string="Leaves to Pro." name="approved_leaves_count" widget="statinfo"/>
```
Comment (tde-banana-odoo) [inline addons/hr_holidays/hr_holidays_view.xml]: Leaves to Pro. ?

File: addons/hr_holidays/hr_holidays_view.xml [hunk 2]
```diff
@@ -587,19 +587,22 @@
...
                     </button>
                     <button name="%(act_hr_employee_holiday_request)d"
                             type="action"
                             class="oe_stat_button"
                             icon="fa-calendar"
-                            groups="base.group_hr_user">
-                        <field name="leaves_count" widget="statinfo"/>
+                            attrs="{'invisible': [('show_leaves','=', False)]}"
+                            groups="base.group_user" help="Remaining leaves">
+                        <field string="Remaining Lea." name="leaves_count" widget="statinfo"/>
```
Comment (tde-banana-odoo) [inline addons/hr_holidays/hr_holidays_view.xml]: Remaining Lea. ?

File: addons/hr_holidays/hr_holidays.py [hunk 3]
```diff
@@ -596,9 +596,16 @@ def _leaves_count(self, cr, uid, ids, field_name, arg, context=None):
...
         for employee_id in ids:
-            leaves = Holidays.search_count(cr, uid, [('employee_id', '=', employee_id), ('type', '=', 'remove')], context=context)
-            approved_leaves = Holidays.search_count(cr, uid, [('employee_id', '=', employee_id), ('type', '=', 'remove'), ('date_from', '>=', date_begin.strftime(tools.DEFAULT_SERVER_DATE_FORMAT)), ('date_from', '<=', date_end.strftime(tools.DEFAULT_SERVER_DATE_FORMAT)), ('state', '=', 'validate'), ('payslip_status', '=', False)], context=context)
-            res[employee_id] = {'leaves_count': leaves, 'approved_leaves_count': approved_leaves}
+            leaves = Holidays.read_group(cr, uid, [('employee_id', '=', employee_id), ('holiday_status_id.limit', '=', False)], fields=['number_of_days'], groupby=[], context=context)
+            approved_leaves = Holidays.search_count(cr, uid, [('employee_id', '=', employee_id), ('type', '=', 'remove'), ('date_from', '>=', date_begin.strftime(tools.DEFAULT_SERVER_DATE_FORMAT)), ('date_from', '<=', date_end.strftime(tools.DEFAULT_SERVER_DATE_FORMAT)), ('payslip_status', '=', False)], context=context)
```
Comment (tde-banana-odoo) [inline addons/hr_holidays/hr_holidays.py]: There is a mismatch, either you rename the field, either removing the validate state is not correct.

Comment (bba-odoo) [inline addons/hr_holidays/hr_holidays.py]: ok, i rename this field.

File: addons/hr_holidays/hr_holidays_view.xml (hunk 1 above)
Comment (bba-odoo) [inline addons/hr_holidays/hr_holidays_view.xml]: string is larger then button fix width (in oe_state button). So I use this way reference from following.


I is this not okay then can I write full string, but string overflow going hidden ? Thanks
download

File: addons/hr_holidays/hr_holidays.py [hunk 4]
```diff
@@ -596,9 +596,16 @@ def _leaves_count(self, cr, uid, ids, field_name, arg, context=None):
...
-            res[employee_id] = {'leaves_count': leaves, 'approved_leaves_count': approved_leaves}
+            leaves = Holidays.read_group(cr, uid, [('employee_id', '=', employee_id), ('holiday_status_id.limit', '=', False)], fields=['number_of_days'], groupby=[], context=context)
+            leaves_to_process = Holidays.search_count(cr, uid, [('employee_id', '=', employee_id), ('type', '=', 'remove'), ('date_from', '>=', date_begin.strftime(tools.DEFAULT_SERVER_DATE_FORMAT)), ('date_from', '<=', date_end.strftime(tools.DEFAULT_SERVER_DATE_FORMAT)), ('payslip_status', '=', False)], context=context)
+            res[employee_id] = {'leaves_count': leaves[0]['number_of_days'], 'leaves_to_processe_count': leaves_to_process}
+        return res
+
+    def _show_approved_remaining_leave(self, cr, uid, ids, name, args, context=None):
+        res={}
+        employee= self.browse(cr, uid, ids, context=context)
```
Comment (tde-banana-odoo) [inline addons/hr_holidays/hr_holidays.py]: This won't work in batch. Remember you receive ids that can be a list containing more than one element. Here you should do
 - if user is hr: return a dict with all ids to true
- check user's employee, set this id to true, other to false

File: addons/hr_holidays/hr_holidays.py [hunk 5]
```diff
@@ -596,9 +596,16 @@ def _leaves_count(self, cr, uid, ids, field_name, arg, context=None):
         date_begin = date.today().replace(day=1)
         date_end = date_begin.replace(day=calendar.monthrange(date_begin.year, date_begin.month)[1])
         for employee_id in ids:
-            leaves = Holidays.search_count(cr, uid, [('employee_id', '=', employee_id), ('type', '=', 'remove')], context=context)
-            approved_leaves = Holidays.search_count(cr, uid, [('employee_id', '=', employee_id), ('type', '=', 'remove'), ('date_from', '>=', date_begin.strftime(tools.DEFAULT_SERVER_DATE_FORMAT)), ('date_from', '<=', date_end.strftime(tools.DEFAULT_SERVER_DATE_FORMAT)), ('state', '=', 'validate'), ('payslip_status', '=', False)], context=context)
-            res[employee_id] = {'leaves_count': leaves, 'approved_leaves_count': approved_leaves}
+            leaves = Holidays.read_group(cr, uid, [('employee_id', '=', employee_id), ('holiday_status_id.limit', '=', False)], fields=['number_of_days'], groupby=[], context=context)
```
Comment (tde-banana-odoo) [inline addons/hr_holidays/hr_holidays.py]: Why doing a loop for read_group ? Moreover the second search_count could be implemented using a read_group. This should allow you to remove the loop.

File: addons/hr_holidays/hr_holidays_view.xml [hunk 6]
```diff
@@ -587,19 +587,22 @@
...
+                    <field name="show_leaves" invisible="1"/>
                     <button name="%(act_hr_employee_holiday_request_approved)d"
                             icon="fa-calendar"
                             class="oe_stat_button"
                             type="action"
-                            groups="base.group_hr_user">
-                        <field name="approved_leaves_count" widget="statinfo"/>
+                            attrs="{'invisible': [('show_leaves','=', False)]}"
+                            groups="base.group_user" help="Leaves to Process">
+                        <field string="Leaves to Pro." name="leaves_to_process_count" widget="statinfo"/>
```
Comment (tde-banana-odoo) [inline addons/hr_holidays/hr_holidays_view.xml]: Use a full name or find a shortcut, but avoid strings meaning nothing.

File: addons/hr_holidays/hr_holidays_view.xml (hunk 2 above)
Comment (tde-banana-odoo) [inline addons/hr_holidays/hr_holidays_view.xml]: Use something more meaningful.

File: addons/hr_holidays/hr_holidays.py (hunk 5 above)
Comment (bba-odoo) [inline addons/hr_holidays/hr_holidays.py]: done.

File: addons/hr_holidays/hr_holidays.py (hunk 4 above)
Comment (bba-odoo) [inline addons/hr_holidays/hr_holidays.py]: Done.

File: addons/hr_holidays/hr_holidays_view.xml (hunk 6 above)
Comment (bba-odoo) [inline addons/hr_holidays/hr_holidays_view.xml]: Done.

**Comment by antonylesuisse** [comment]
Cancelled task.

**Pull request closed without merging**